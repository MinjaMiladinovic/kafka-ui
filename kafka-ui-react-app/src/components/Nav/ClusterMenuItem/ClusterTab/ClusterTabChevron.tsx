import React from 'react';

interface Props {
  isOpen: boolean;
}

const ClusterTabChevron: React.FC<Props> = ({ isOpen }) => {
  return (
    <svg
      className="cluster-tab-chevron"
      width="10"
      height="6"
      viewBox="0 0 10 6"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d={isOpen ? 'M8.99988 5L4.99988 1L0.999878 5' : 'M1 1L5 5L9 1'}
        stroke="#73848C"
      />
    </svg>
  );
};

export default ClusterTabChevron;
