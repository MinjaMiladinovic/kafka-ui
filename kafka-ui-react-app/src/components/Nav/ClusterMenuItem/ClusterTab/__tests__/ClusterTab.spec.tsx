import { render } from '@testing-library/react';
import ClusterTab, {
  ClusterTabProps,
} from 'components/Nav/ClusterMenuItem/ClusterTab/ClusterTab';
import { ServerStatus } from 'generated-sources';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from 'theme/theme';

describe('ClusterTab component', () => {
  const setupWrapper = (props?: Partial<ClusterTabProps>) => (
    <ThemeProvider theme={theme}>
      <ClusterTab
        to="/test"
        status={ServerStatus.ONLINE}
        isOpen
        toggleClusterMenu={jest.fn()}
        {...props}
      />
    </ThemeProvider>
  );
  describe('with default props', () => {
    it('matches the snapshot', () => {
      const component = render(setupWrapper());
      expect(component.baseElement).toMatchSnapshot();
    });
  });
});
