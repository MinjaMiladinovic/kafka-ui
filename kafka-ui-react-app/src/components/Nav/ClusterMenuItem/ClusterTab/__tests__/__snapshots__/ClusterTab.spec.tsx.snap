// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterTab component with default props matches the snapshot 1`] = `
<body>
  <div>
    <li
      class="sc-bdfBQB dqCete"
    >
      <div
        class="cluster-tab-wrapper"
      >
        <div
          class="cluster-tab-l"
        >
          <svg
            fill="none"
            height="4"
            viewBox="0 0 4 4"
            width="4"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="2"
              cy="2"
              fill="#5CD685"
              r="2"
            />
          </svg>
        </div>
        <svg
          class="cluster-tab-chevron"
          fill="none"
          height="6"
          viewBox="0 0 10 6"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.99988 5L4.99988 1L0.999878 5"
            stroke="#73848C"
          />
        </svg>
      </div>
    </li>
  </div>
</body>
`;
